week5.2-ex2.py

```diff
@@ -0,0 +1,33 @@
+from functools import reduce
+
+
+def add_floats(number1, number2):
+    """Sum 2 floats- the function sum() working with 2 iterables"""
+    return number1+number2
+
+
+def get_recipe_price(prices, optionals=None, **ingredients_and_amounts):
+    """Return the price of all the ingredients in the recipe according the prices and per 100 gram"""
+    if optionals is None:
+        optionals = []
+    #  export list that contain all the ingredients for the recipe and the price for the specific amount
+    ingredients = [price*(amount/100) for ingredient1, price in prices.items() if ingredient1 not in optionals for
+                   ingredient2, amount in ingredients_and_amounts.items() if ingredient1 == ingredient2]
+    if not ingredients:  # if there is no ingredients for the recipe
```

Comment:
when will this condition evaluate to true? when ingredients will be None? I want ot make sure you you understand it

Reply:
When the ingredients that are optionals and we need to ignore from them are the same ingredients like the dictionary with the ingredients for the recipe.